File: Make Human Unity/Assets/MakeHuman/Icons/KsSoft/Editor/Multilingual/tools/protocol/t_SoundEffect.py
from CVariable import CReadVariable,CWriteVariable
from Vector import *
#/==========================================================================
#*!
#	@brief	t_SoundEffect
#/
class t_SoundEffect:
	#/==========================================================================
	#*!
	#	@brief	Member
	#/
	m_id = 0	#U32
	m_mAssetBundle = 0	#U32
	m_idClip = 0	#U32
	#/==========================================================================
	#*!
	#	@brief	Constructor
	#/
	def __init__(self):
		self.clear()

	#/==========================================================================
	#*!
	#	@brief	Accessor
	#/
	def clear(self):
		self.m_id = 0
		self.m_mAssetBundle = 0
		self.m_idClip = 0

	def read(self,cVariable):
		self.m_id = cVariable.getU32()
		self.m_mAssetBundle = cVariable.getU32()
		self.m_idClip = cVariable.getU32()
		return True

	def write(self,cVariable):
		cVariable.putU32(self.m_id)
		cVariable.putU32(self.m_mAssetBundle)
		cVariable.putU32(self.m_idClip)
		return True

File: Assets/MakeHuman/Icons/KsSoft/Plugins/Protocol/t_SoundEffect.cs
//==============================================================================================
/*!
	@file  KsSoft Protocol
	
	(counter SJIS string 京)
*/
//==============================================================================================
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

using KS;
//==========================================================================
/*!
	@brief	t_SoundEffect
*/
public class t_SoundEffect : ISerializable {
	//==========================================================================
	/*!
		@brief	Member
	*/
	public uint	m_id;
	public uint	m_mAssetBundle;
	public uint	m_idClip;
	//==========================================================================
	/*!
		@brief	Constructor
	*/
	public t_SoundEffect() {
		clear();
	}
	//==========================================================================
	/*!
		@brief	Accessor
	*/
	public void clear() {
		m_id = 0;
		m_mAssetBundle = 0;
		m_idClip = 0;
	}
	public bool read(CReadVariable cVariable) {
		try {
			m_id = cVariable.getU32();
			m_mAssetBundle = cVariable.getU32();
			m_idClip = cVariable.getU32();
		} catch (System.Exception e) {
			Debug.LogError(e);return false;
		}
		return true;
	}
	public bool write(CWriteVariable cVariable) {
		cVariable.put(m_id);
		cVariable.put(m_mAssetBundle);
		cVariable.put(m_idClip);
		return true;
	}
};

